data-semantics.md

```diff
@@ -0,0 +1,106 @@
+---
+layout: page
+title: Data Semantics Guidelines
+---
+<div id="toc"></div>
+
+## Introduction
+
+OpenTracing defines an API through which application instrumentation can log data to a pluggable tracer. In general, there is no guarantee made by OpenTracing about the way that data will be handled by an underlying tracer. That said, the APIs are by design open-ended: what type of data should be provided to them?
+
+A general understanding between the instrumentor and tracer developers adds great value: if certain known tag key/values are used for common application scenarios, tracers can choose to pay special attention to them. The same is true of `log`ged events, and span structure in general.
+
+As an example, consider the common case of a HTTP-based application server. The URL of an incoming request that the application is handling is often useful for diagnostics, as well as the HTTP verb and the resultant status code. An instrumentor could choose to report the URL in a tag named `URL`, or perhaps named `http.url`--either would be valid from the pure API perspective. But if the Tracer wishes to add intelligence, such as indexing on the URL value or sampling proactively for requests to a particular endpoint, it must know where to look for relevant data. In short, when tag names and other instrumentor-provided values are used consistently, the tracers on the other side of the API can employ more intelligence.
+
+The guidelines contained describe a common ground on which instrumentors and tracer authors can build beyond pure data collection. Adherence to the guidelines is optional but highly recommended for instrumetors.
+
+
+
+## Spans
+
+
+### Span Naming
+
+Spans carry tags, logs, and attributes, but they also have a top-level **operation name**. This should be a low-cardinality string value representing the type of work being done in the span. Typically, a process, framework, library, or module name is a good guideline.
+
+For instance, a request being handled by a Flask-based application might be instrumented using a WSGI middleware. In this situation, the first span operation name might be `WSGI`. If the application subsequently makes a database query, the span name might be `SQL`.
```

Comment:
I think operation is valuable, but it seems to miss a step in the pyramid--the step that defines what exactly is doing the operating.  There's certain things we are expecting we can provide automatically via traces, like hostname, which will help localize where work is being done.  Component feels as though it ought to be one, though it's not one that I would expect most tracers to be able to fetch automatically as hostname, so we would expect users to provide in the same way they provide op.

@yurishkuro Is `component_name` a standardized thing?  I don't see it in any of the codebases, but I do think it makes sense as a label for this.  Potentially just `component` as well.  One issue with your Redis example is I could see the same operation names being used for memcache, or HTTP--there's a bit of ambiguity if you skip the `component` element.

@adriancole I agree that more precision must be provided.  There is some UI bias inherent in terms of how we're thinking about this (from myself included!).

At root, the issue is that a single attribute of a span (`operation_name`) has been elevated to the top level, and everything else is subjugated as tags.  I expect in many cases the `operation_name` ought to be supplemented by further tags.  OT will benefit from being as precise as possible about what is a good type of value for this top-level attribute vs the other tags.

So I'm fine with trying to keep `operation_name` as some sort of operation name, and adding a concept of `component` separately, but conceptually it feels clearer to have the broader-strokes `component` as our elevated attribute.  Either way, we will want to tighten this up.  Could be as simple as a set of examples and some guidelines (eg. I think "low cardinality" is a good one)?